### PR TITLE
S3, GCS Destinations : Fix interface duplication

### LINKED ISSUE #8894: Merge S3Writer, GcsWriter, CommonWriter interfaces
We have three very similar interfaces.
Find way to merge them.

### PR DESCRIPTION
**Pull request by DoNotPanicUA**
## What
#8894
Remove logical duplication from the writer interfaces.
Is blocked by #6063

## How
Create new generic interfaces at the bottom (S3 destination is the lowest level right now for writers). Remove duplication interfaces from the GCS destination.

## Recommended reading order
1. `DestinationFileWriter.java`
2. `DestinationWriter.java`

### CODE REVIEW

File: airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public void useInsertStrategyTest() {
     assertFalse(SnowflakeDestination.isS3Copy(stubConfig));
   }
 
-  @Test
+  //@Test
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java:76]: are you going to uncomment this test?

Comment (mkhokh-33) [reply airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationTest.java:76]: please check do we need this test?

**Review by mkhokh-33 — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/writer/BaseS3Writer.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
  * <li>Log and close the write.</li>
  * </ul>
  */
-public abstract class BaseS3Writer implements S3Writer {
+public abstract class BaseS3Writer implements DestinationFileWriter {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(BaseS3Writer.class);
   public static final String DEFAULT_SUFFIX = "_0";
```
Comment (mkhokh-33) [inline airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/writer/BaseS3Writer.java:40]: I guess we could change it to private as constant is used only inside this class :)

**Review by mkhokh-33 — CHANGES_REQUESTED** [review]

**Review by alexandertsukanov — APPROVED** [review]
No critical remarks from my side.

**Review by mkhokh-33 — APPROVED** [review]

**Review by alexandr-shegeda — APPROVED** [review]

**Pull request merged**